app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -25,7 +25,8 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:intl/intl.dart' show toBeginningOfSentenceCase;
 import 'package:mime/mime.dart';
 
-final mentionKey = GlobalKey<FlutterMentionsState>();
+// keep track of text controller values across rooms.
+final _textValuesProvider = StateProvider<Map<String, String>>((ref) => {});
```

Comment:
Why use a MapState rather than just a state family?

```suggestion
final _textValuesProvider = StateProvider.family<String, String>((ref) => '');
```